catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ValidateCommand.java

```diff
@@ -47,178 +55,104 @@
     @Completion(FileCompleter.class)
-    String filename;
+    String path;
+
+    @Option(name = "--query", aliases = "-q", description = "Text query for catalog entries to validate")
+    String query;
 
     @Reference
     List<MetacardValidator> validators;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ValidateCommand.class);
-
-    // Using system out for println is caught by our githooks to prevent bad practice.
-    // We actually need to use it here.
-    protected PrintStream console = System.out;
-
-    // These constants are to help in trying to make the output
-    // both uniform and readable.
-    private static final String INDENT = "|   ";
-
-    private static final String LAST_INDENT = "    ";
-
-    private static final String ELEMENT = "+-- ";
-
-    private static final String LAST_ELEMENT = "`-- ";
-
-    private static final String WARNING_TITLE = "Warnings";
-
-    private static final String ERROR_TITLE = "Errors";
-
-    private static final String WARNING_COLOR = COLOR_CYAN;
+    @Reference
+    CatalogFramework catalog;
 
-    private static final String ERROR_COLOR = COLOR_RED;
+    @Reference
+    FilterBuilder builder;
 
     @Override
     public Object execute() throws Exception {
-
-        if (fileExists() && hasValidators()) {
-            Metacard metacard = createMetacard();
-            if (metacard != null) {
-                runValidators(metacard);
+        int bad = 0;
+        if (validators != null && validators.size() > 0) {
+            List<Metacard> metacards;
+            if (path != null) {
+                metacards = createMetacards();
+            } else if (query != null) {
+                metacards = getMetacardsFromCatalog();
+            } else {
+                ValidatePrinter.printError(
+                        "Usage: catalog:validate < --path filePath > < --query textQuery >");
+                return null;
             }
-        }
-
-        // We must return an object per the Action contract which
-        // this class implements. It is sufficient to return null.
-        return null;
-    }
-
-    private boolean fileExists() {
-        File fileToValidate = new File(filename);
-        if (!fileToValidate.exists()) {
-            console.printf(ERROR_COLOR +
-                    "Unable to locate file '%s'. Double check the path is correct and the file exists.%n"
-                    + COLOR_DEFAULT, filename);
-            return false;
-        }
 
-        if (!fileToValidate.isFile()) {
-            console.printf(ERROR_COLOR + "'%s' is not a file.%n" + COLOR_DEFAULT, filename);
-            return false;
+            List<ValidateReport> reports = ValidateExecutor.execute(metacards, validators);
+            for (ValidateReport report : reports) {
+                if (report.getEntries()
+                        .size() > 0) {
+                    bad++;
```

Comment:
It's counting the number of files that had errors or warnings, not the total number of errors or warnings. That's why we increment by one. I should give it a more descriptive variable name